### ISSUE TITLE
Matrix functions for UniformScaling

### ISSUE THREAD

**Issue by jagot** [JuliaLang/julia#34799]
I was surprised by this:
```julia
julia> sqrt(5.0I)
ERROR: MethodError: no method matching sqrt(::UniformScaling{Float64})
Closest candidates are:
  sqrt(::Float16) at math.jl:1085
  sqrt(::Complex{Float16}) at math.jl:1086
  sqrt(::Missing) at math.jl:1138
  ...
Stacktrace:
 [1] top-level scope at REPL[25]:1

julia> sqrt(Matrix(5I, 5, 5))
5×5 Array{Float64,2}:
 2.23607  0.0      0.0      0.0      0.0
 0.0      2.23607  0.0      0.0      0.0
 0.0      0.0      2.23607  0.0      0.0
 0.0      0.0      0.0      2.23607  0.0
 0.0      0.0      0.0      0.0      2.23607
```
Since we already have a few [matrix functions](https://en.wikipedia.org/wiki/Matrix_function) defined, such as
https://github.com/JuliaLang/julia/blob/e36fe95a639c6956aa39367bef8e727b174e52b2/stdlib/LinearAlgebra/src/dense.jl#L702-L759
it would seem reasonable to expect at least these elementary functions to be defined for `UniformScaling` as well:
- `^(::UniformScaling, ::Number)`
- `exp`
- `sin`, `cos`, `tan` + inverses and hyperbolic versions
- `log`
- `sqrt`

**Comment by jagot** [comment]
Initial implementation: https://github.com/jagot/UniformScalingMatrixFunctions.jl

**Comment by dkarrasch** [comment]
There was some initial effort made in #28872, but then got stale.

**Comment by jagot** [comment]
Good to know! Maybe that PR should be revived instead?

**Comment by dkarrasch** [comment]
Yes. Actually, that PR had all that you are asking for here. So we could resolve merge conflicts there, merge, and open a new issue to track the ambitious unresolved broadcasting aspects.

**Comment by JeffreySarnoff** [comment]
> sqrt 
    exp, log
    sin, cos, tan + asin, acos, atan
    sinh, cosh, tanh + asinh, acosh, atanh

other similar from Base.Math:  `cbrt` , `expm1`, `log1p`

**Comment by jagot** [comment]
I've rebased #28872 and pushed to [my fork](https://github.com/jagot/julia/tree/I-methods). Just need to build and test and then make a new PR?

**Comment by dkarrasch** [comment]
I've just resolved the merge conflicts there and one qualification issue, tests are running. If they pass, I'd suggest to merge #28872, and then continue from there.

**Comment by jagot** [comment]
Hmm, double work double the fun?

**Comment by jagot** [comment]
@antoine-levitt suggested this list of functions:
https://github.com/stevengj/ChangePrecision.jl/blob/402b3eec7710552b655555e8335faca2737235e4/src/ChangePrecision.jl#L33-L46
supplemented by
https://github.com/JuliaDiff/DiffRules.jl/blob/master/src/rules.jl

**Comment by dkarrasch** [comment]
> @antoine-levitt suggested this list of functions:
> ...
> supplemented by
> ...

Sounds good. Maybe update the OP to reflect the missing parts now that #28872 is merged? One open issue there were matrix decompositions.

**Comment by jagot** [comment]
Updated!
